File: frontend/src/api/titleCrops.ts
import { apiPost } from "./client";

export type TitleCropBBox = {
  x: number;
  y: number;
  width: number;
  height: number;
};

export type TitleCropResult = {
  sheet_id: number;
  status: string;
  title_crop_path: string | null;
  title_crop_bbox: TitleCropBBox | null;
  error_code: string | null;
  error_message: string | null;
};

export type BatchTitleCropResult = {
  batch_id: number;
  total_count: number;
  success_count: number;
  failed_count: number;
  items: TitleCropResult[];
};

const LONG_OP_TIMEOUT_MS = 300_000;

export function cropSheetTitle(sheetId: number): Promise<TitleCropResult> {
  return apiPost<TitleCropResult>(`/api/sheets/${sheetId}/title-crop`, undefined, {
    timeoutMs: LONG_OP_TIMEOUT_MS
  });
}

export function cropBatchTitles(batchId: number): Promise<BatchTitleCropResult> {
  return apiPost<BatchTitleCropResult>(`/api/imports/${batchId}/title-crops`, undefined, {
    timeoutMs: LONG_OP_TIMEOUT_MS
  });
}

File: backend/api/title_crops.py
from fastapi import APIRouter, Depends
from fastapi.responses import FileResponse
from sqlalchemy.orm import Session

from backend.core.database import get_db
from backend.schemas.drawing_sheet import BatchTitleCropResult, TitleCropResult
from backend.services import title_crop_service

router = APIRouter(prefix="/api", tags=["title-crops"])


@router.post("/sheets/{sheet_id}/title-crop", response_model=TitleCropResult)
def crop_sheet_title(sheet_id: int, db: Session = Depends(get_db)) -> TitleCropResult:
    return title_crop_service.crop_title_block_for_sheet(db, sheet_id)


@router.post("/imports/{batch_id}/title-crops", response_model=BatchTitleCropResult)
def crop_batch_titles(batch_id: int, db: Session = Depends(get_db)) -> BatchTitleCropResult:
    return title_crop_service.crop_title_blocks_for_batch(db, batch_id)


@router.get("/sheets/{sheet_id}/title-crop")
def get_sheet_title_crop(sheet_id: int, db: Session = Depends(get_db)) -> FileResponse:
    path = title_crop_service.title_crop_path(db, sheet_id)
    return FileResponse(path, media_type="image/png")
